x-pack/plugins/lens/public/editor_frame_service/editor_frame/config_panel/layer_actions/clone_layer_action.tsx

```diff
@@ -23,7 +24,7 @@ export const getCloneLayerAction = (props: CloneLayerAction): LayerAction => {
   return {
     execute: props.execute,
     displayName,
-    isCompatible: Boolean(props.activeVisualization.cloneLayer),
+    isCompatible: Boolean(props.activeVisualization.cloneLayer && !props.isTextBasedLanguage),
```

Comment:
ℹ️ clone layer and layer should not be allowed for text based languages (for the initial phase)